SWADroid/src/main/res/layout/button_quadrant.xml

```diff
@@ -0,0 +1,208 @@
+                android:orientation="horizontal"
+                android:visibility="gone">
+
+                <ToggleButton
+                    android:id="@+id/button1"
+                    android:layout_width="0dp"
+                    android:layout_height="match_parent"
+                    android:layout_margin="5dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/togglebuttona"
+                    android:text='a'
+                    android:textAllCaps="false"
+                    android:textColor="#ffffff"
+                    android:textOff='a'
+                    android:textOn='a'
+                    android:textSize="30dp" />
+
+                <ToggleButton
+                    android:id="@+id/button2"
+                    android:layout_width="0dp"
+                    android:layout_height="match_parent"
+                    android:layout_margin="5dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/togglebuttonb"
+                    android:text='b'
+                    android:textAllCaps="false"
+                    android:textColor="#ffffff"
+                    android:textOff='b'
+                    android:textOn='b'
+                    android:textSize="30dp" />
+            </LinearLayout>
+
+            <LinearLayout
+                android:id="@+id/linearLayout34"
+                android:layout_width="match_parent"
+                android:layout_height="0dp"
+                android:layout_weight="1"
+                android:orientation="horizontal"
+                android:visibility="gone">
+
+                <ToggleButton
+                    android:id="@+id/button3"
+                    android:layout_width="0dp"
+                    android:layout_height="match_parent"
+                    android:layout_margin="5dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/togglebuttonc"
+                    android:text='c'
+                    android:textAllCaps="false"
+                    android:textColor="#ffffff"
+                    android:textOff='c'
+                    android:textOn='c'
+                    android:textSize="30dp" />
+
+                <ToggleButton
+                    android:id="@+id/button4"
+                    android:layout_width="0dp"
+                    android:layout_height="match_parent"
+                    android:layout_margin="5dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/togglebuttond"
+                    android:text='d'
+                    android:textAllCaps="false"
+                    android:textColor="#ffffff"
+                    android:textOff='d'
+                    android:textOn='d'
+                    android:textSize="30dp" />
+            </LinearLayout>
+
+            <LinearLayout
+                android:id="@+id/linearLayout56"

```

Comment:
You should use a more descriptive name for the id. This improves the readability of the code that refers to this id and makes it easier to identify which button is being used.